### PR TITLE
JENKINS-59009 : Adding Proxy Support

### PR DESCRIPTION
**Pull request by flocard**
# Description

Using Jenkins Proxy Configuration if exist in OkHttpClient
See [JENKINS-59009](https://issues.jenkins-ci.org/browse/JENKINS-59009).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description.
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests.
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description.
- [ ] Reviewed the code.
- [ ] Verified that the appropriate tests have been written or valid explanation given.
- [ ] If applicable, tested by installing this plugin on the Jenkins instance.

### CODE REVIEW

File: src/main/java/org/thoughtslive/jenkins/plugins/jira/service/JiraService.java [hunk 1]
```diff
@@ -38,11 +40,28 @@ public JiraService(final Site jiraSite) {
 
     final ConnectionPool CONNECTION_POOL = new ConnectionPool(5, 60, TimeUnit.SECONDS);
 
-    OkHttpClient httpClient = new OkHttpClient.Builder()
-        .connectTimeout(jiraSite.getTimeout(), TimeUnit.MILLISECONDS)
-        .readTimeout(jiraSite.getReadTimeout(), TimeUnit.MILLISECONDS)
-        .connectionPool(CONNECTION_POOL)
-        .retryOnConnectionFailure(true).addInterceptor(new SigningInterceptor(jiraSite)).build();
+    OkHttpClient.Builder OkHttpClientBuilder = new OkHttpClient.Builder()
```
Comment (nrayapati) [inline src/main/java/org/thoughtslive/jenkins/plugins/jira/service/JiraService.java]: nit: Can we rename the variable to follow the java naming conventions, just lowercase O

**Comment by flocard** [comment]
> @flocard Thank you very much making this PR, hope you have tested this change? as I don't have a instance where I can configure http proxy. Thanks again.

Hi,
Yes the PR is tested, I currently use this modified plugin on my Jenkins Instance which doesn't have direct access to my Jira Instance.

**Comment by guija** [comment]
Hi. Since we updated to Version 1.5.4 which includes this change our setup does not work anymore. Where exactly have the proxy settings to be configured?
Our setup:
- Jenkins and Jira are in the same network.
What I've tried:
- Added the site to the ignored site for the proxy in the proxy settings in the Jenkins settings ( Jenkins - Manage Jenkins - Manage Plugins Advanced). 
- Set environment variables (HTTP_PROXY, HTTPS_PROXY and especially **NO_PROXY**).

Do you have any hints?

**Comment by flocard** [comment]
Hi,

I create a Jira issue https://issues.jenkins-ci.org/browse/JENKINS-62193 and making a pull request https://github.com/jenkinsci/jira-steps-plugin/pull/108 which adding no proxy host config

**Pull request merged**